### PR TITLE
Types extraction

### PR DESCRIPTION
**Pull request by LeonY1**
Making changes to the types of each directory and placing them into the const and types directory.

Solves #199

### CODE REVIEW

**Comment by fyu** [comment]
Thanks for your contribution! Please make sure to pass lint and tests. You may ignore s3_storage test because you probably don't have your aws set up.

**Comment by LeonY1** [comment]
@fyu what is the best node version to test on? When I ran the `npm run test` and `npm run lint`, I solved all the linting issues and only had the s3 issue.

**Comment by fyu** [comment]
In github workflow, we test different node versions: https://github.com/scalabel/scalabel/actions/runs/186044206

**Comment by LeonY1** [comment]
How do I get past the AWS_Config issue from the `npm run test`?

File: app/src/js/types/common.ts [hunk 1]
```diff
@@ -1,5 +1,27 @@
+import { ActionCreator, AnyAction, Store } from 'redux'
+import { ThunkAction, ThunkDispatch } from 'redux-thunk'
+import { StateWithHistory } from 'redux-undo'
+import { BaseAction } from '../types/action'
+import { State } from '../types/functional'
```
Comment (fyu) [inline app/src/js/types/common.ts:5]: We can change `types/functional` to `types/state`

File: app/src/js/types/common.ts [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
 
 /**
  * Defining the types of some general callback functions
  */
 export type MaybeError = Error | null | undefined
+
+/**
```
Comment (fyu) [inline app/src/js/types/common.ts:12]: The types below can be moved to `types/redux`

**Comment by fyu** [comment]
> How do I get past the AWS_Config issue from the `npm run test`?

After PR #202, you can use your own AWS keys to test the S3 storage. You can put the keys in github secrets of your fork: https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets. The two secrets are `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.

I have tested your code on my local machines. Lint and tests have passed. The merge will be tested again in the master branch.

**Pull request merged**